### PR TITLE
Merge function

### PR DESCRIPTION
**Pull request by JMurphyWeb**
add comments to merge function

#295

### CODE REVIEW

File: lib/merger.ex [hunk 1]
```diff
@@ -3,18 +3,26 @@ defmodule Karma.Merger do
...
-  #   res
+  #   {:ok, merged_path} = wrap_merge_script(json, doc_path, merged_path)
+  #
+  #
   #   # save to S3
+  #   # get file name from merged path
+  #   image_params = %{path: merged_path, filename: "test-1-2-3.pdf"}
+  #   {:ok, :url, url} = S3.upload({:url, image_params})
   #
   #   # add merged url to document's table
```
Comment (JMurphyWeb) [inline lib/merger.ex]: one of us can take on this last task

File: lib/merger.ex [hunk 2]
```diff
@@ -3,18 +3,26 @@ defmodule Karma.Merger do
...
+  #   # add foreign key offer_id to the document table
+  #   # add this url to the document table
+  # 
+  #    url
```
Comment (JMurphyWeb) [inline lib/merger.ex]: One of us could integrating into controller, if it works deploy to test on Heroku

File: lib/merger.ex [hunk 3]
```diff
@@ -1,21 +1,37 @@
...
-  #   res = wrap_merge_script(json, doc_path, merged_path)
-  #   res
-  #   # save to S3
-  #
-  #   # add merged url to document's table
-  # end
+  def merge(offer, document) do
+    # download document
+
+    case Karma.S3.download(document.url, System.cwd()<>"/test.pdf") do
```
Comment (JMurphyWeb) [inline lib/merger.ex]: here we are hardcoding the file path

File: lib/merger.ex [hunk 4]
```diff
@@ -1,21 +1,37 @@
...
+
+        # do merge
+        merged_path = get_merged_path(doc_path, offer, document)
+        case wrap_merge_script(json, doc_path, merged_path) do
+          {:error, _error} ->
+            {:error, "There was an error creating the document"}
+          {:ok, merged_path} ->
+            # save to S3
+            # get file name from merged path
+            image_params = %{path: merged_path, filename: "test-1-2-3.pdf"}
```
Comment (JMurphyWeb) [inline lib/merger.ex]: again hardcoding the file name

File: web/controllers/offer_controller.ex [hunk 5]
```diff
@@ -237,6 +239,15 @@ defmodule Karma.OfferController do
         if offer.accepted == true do
           Karma.Email.send_offer_accepted_contractor(conn, offer)
           |> Karma.Mailer.deliver_later()
+
+          # Move to document model?
+          [document] = Karma.Repo.all(
+          from d in Karma.Document,
+          where: d.project_id == ^project.id
+          and d.name == "PAYE"
```
Comment (JMurphyWeb) [inline web/controllers/offer_controller.ex]: Needs fixing.

Make sure all contract types are the same.

Limit types of contract.

**Comment by jackcarlisle** [comment]
closing, changes absorbed by #360

**Pull request closed without merging**